lib/hitters.go

```diff
@@ -0,0 +1,209 @@
+package vegeta
+
+import (
+	"bytes"
+	"crypto/tls"
+	"io"
+	"io/ioutil"
+	"math"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/valyala/fasthttp"
+)
+
+// TODO: Go-docs for everything here.
+// TODO: Overhaul examples in README to use new APIs
+// TODO: Fix attacker tests
+// TODO: Write CLI integration tests
+
+type Hitter interface {
+	Hit(*Target) *Result
```

Comment:
Why pointers and not values? I bet you might see some even substantial perf improvements on long test runs if this were

```go
type Hitter interface {
    Hit(Target) Result
}
```

Reply:
Benchmarks seems inconclusive, but slightly worse with values. These aren't very small structs. It follows the same pattern as the `Doer` interface, which takes in an `*http.Request` and returns an `*http.Response`.

Reply:
Old = pointers
New = values

```
name                   old time/op    new time/op    delta
FastHTTPHitter_Hit-12    10.3µs ± 6%    10.7µs ±11%   ~     (p=0.133 n=9+10)
NetHTTPHitter_Hit-12     1.79ms ± 4%    1.79ms ±15%   ~     (p=0.167 n=8+9)

name                   old req/s      new req/s      delta
FastHTTPHitter_Hit-12     95.8k ±12%     93.8k ±10%   ~     (p=0.315 n=10+10)
NetHTTPHitter_Hit-12      3.46 ±700%   117.33 ±372%   ~     (p=0.529 n=8+9)

name                   old alloc/op   new alloc/op   delta
FastHTTPHitter_Hit-12    1.28kB ± 1%    1.28kB ± 1%   ~     (p=0.955 n=10+10)
NetHTTPHitter_Hit-12     5.45kB ±28%    6.21kB ±50%   ~     (p=0.459 n=9+10)

name                   old allocs/op  new allocs/op  delta
FastHTTPHitter_Hit-12      21.0 ± 0%      21.0 ± 0%   ~     (all equal)
NetHTTPHitter_Hit-12       79.0 ± 0%      80.4 ±11%   ~     (p=0.776 n=8+9)
```

Reply:
I think you'd have to run the test for long enough to trigger a bunch of GC in order to see meaningful differences. But no big deal in any case.